src/trace_impls.rs

```diff
@@ -428,8 +434,8 @@ mod sync {
 
     impl<T: Trace> Trace for sync::Mutex<T> {
         fn trace(&self, tracer: &mut Tracer) {
-            let x = self.lock().unwrap();
-            x.trace(tracer);
+            // For the same reason as RefCell
+            self.try_lock().unwrap().trace(tracer);
```

Comment:
Could you update the `unwrap` to provide some extra context? For example:

    self.try_lock().expect("bug: Mutex was locked bypassing ThreadedCcRef").trace(tracer)

Now I think `Cc<Mutex<_>>` should be forbidden. I'll figure out a way to do that.

Reply:
Why `Cc<Mutex<_>>` be forbidden?

Reply:
Actually that is a good question. The reason is for `Cc<Mutex<T>>`, the single-thread collector won't have things like `ThreadedCcRef` to avoid external threads from locking or keeping references of `T`.

Now I actually like the initial solution in #4, since it works for both `Cc` and `ThreadedCc`. So I will take that, and also the comments here. Thanks!